Raster overlay lifecycle and doc improvements

kring opened the pull request:
This PR came out of this community forum post:
https://community.cesium.com/t/remove-clipping-cesiumpolygonrasteroverlay-added-during-runtime/32520/6

It makes it possible to call the `Destroy Component` Blueprint function on a raster overlay from outside the Cesium3DTileset Actor, and have it actually work.

It fixes a small bug where calling `Refresh` on an inactive `CesiumRasterOverlay` would cause that overlay to appear on the tileset. Until the next time the Tileset was refreshed, and then it would disappear.

It also adds some more detailed documentation for the `AddToTileset`, `RemoveFromTileset`, `Refresh`, `Activate`, and `Deactivate` methods on `CesiumRasterOverlay`.

j9liu commented on Source/CesiumRuntime/Public/CesiumRasterOverlay.h:
```diff
@@ -125,8 +137,33 @@ class CESIUMRUNTIME_API UCesiumRasterOverlay : public UActorComponent {
   void SetSubTileCacheBytes(int64 Value);
 
+  /**
+   * Activates this raster overlay, which will add it to (that is: display it
+   * on) the Cesium3DTileset to which the component is attached, if it hasn't
+   * already been added. The overlay will continue to be shown on the tileset
+   * until it is deactivated.
```
Maybe this is a personal preference, but I think flipping the verbiage here relates this to `AddToTileset` a bit more clearly

```suggestion
   * Activates this raster overlay, which will display it on (that is: add it to)
   * the Cesium3DTileset to which the component is attached, if it isn't
   * already displayed. The overlay will continue to be shown on the
   * tileset until it is deactivated.
```

j9liu commented on Source/CesiumRuntime/Public/CesiumRasterOverlay.h:
```diff
@@ -125,8 +137,33 @@ class CESIUMRUNTIME_API UCesiumRasterOverlay : public UActorComponent {
+   *
+   * If the overlay is already displayed on the Cesium3DTileset, calling this
+   * function will not cause it to pick up any new values for properties that
+   * have been modified since it as added. To do that, call Refresh.
```
```suggestion
   * have been modified since it was added. To do that, call Refresh.
```

j9liu commented on Source/CesiumRuntime/Public/CesiumRasterOverlay.h:
```diff
@@ -125,8 +137,33 @@ class CESIUMRUNTIME_API UCesiumRasterOverlay : public UActorComponent {
   virtual void Activate(bool bReset) override;
+
+  /**
+   * Deactivates this raster overlay. This will remove it from (that is: stop
+   * displaying it on) the Cesium3DTileset to which the component is attached.
```
```suggestion
   * Deactivates this raster overlay. This will stop displaying it on (that is:
   * remove it from) the Cesium3DTileset to which the component is attached.
```

j9liu reviewed (commented):
Looks good @kring , just some minor doc nitpicks. Happy to merge otherwise!

kring commented:
Thanks @j9liu, I've addressed all your feedback. In the doc for the Activate / Deactivate methods, I just removed the "add it to" / "remove it from" bits completely. They weren't adding much and the parenthetical was awkward.

j9liu commented:
Thanks @kring ! Merging now :smile:
